src/scenes/ObservationsRouter/BiomassMeasurementsDetails.tsx

```diff
@@ -138,8 +141,65 @@ export default function BiomassMeasurementsDetails(): JSX.Element {
     (photo) => photo.type === 'Plot' && photo.position !== undefined && photo.position !== 'SouthwestCorner'
   );
 
+  const downloadCsv = async (
+    fileNameSuffix: string,
+    fetchContent: (observationId: number) => Promise<string | null>
+  ) => {
+    if (observation) {
+      const content = await fetchContent(observation.observationId);
+
+      if (content != null && plantingSite != null) {
+        const fileName = sanitize(`${plantingSite.name}-${observation?.startDate}-${fileNameSuffix}.csv`);
+
+        const encodedUri = `data:text/csv;charset=utf-8,` + encodeURIComponent(content);
+
+        const link = document.createElement('a');
+        link.setAttribute('href', encodedUri);
+        link.setAttribute('download', fileName);
+        link.click();
+      }
+    }
+  };
+
+  const exportDetailsCsv = useCallback(async () => {
+    await downloadCsv(strings.BIOMASS_MEASUREMENTS, ObservationsService.exportBiomassDetailsCsv);
+  }, [observation]);
+
+  const exportSpeciesCsv = useCallback(async () => {
+    await downloadCsv(strings.SPECIES, ObservationsService.exportBiomassSpeciesCsv);
+  }, [observation]);
+
+  const exportTreesShrubsCsv = useCallback(async () => {
+    await downloadCsv(strings.TREES_AND_SHRUBS, ObservationsService.exportBiomassTreesShrubsCsv);
+  }, [observation]);
```

Comment:
I think these will need to also be dependent on `plantingSite`, otherwise it might always be null.